### PR TITLE
run: Add the possibility to load an env file

### PR DESCRIPTION
**Pull request by elcore**
This PR adds the possibility to load an env file from the `run` command. 

`caddy run --envfile <path>`

Credit: @SmilingNavern for implementing this code in Caddy v1

### CODE REVIEW

**Comment by francislavoie** [comment]
Closes https://github.com/caddyserver/caddy/issues/3009

File: cmd/main.go [hunk 1]
```diff
@@ -339,6 +340,73 @@ func flagHelp(fs *flag.FlagSet) string {
...
+		}
+
+		// skip empty line
+		if len(line) == 0 {
+			continue
+		}
+
+		fields := strings.SplitN(line, "=", 2)
+		if len(fields) != 2 {
+			return nil, fmt.Errorf("Can't parse line %d; line should be in KEY=VALUE format", lineNumber)
```
Comment (mholt) [inline cmd/main.go:387]: I like these errors; but they should not start with a capital letter
```suggestion
			return nil, fmt.Errorf("can't parse line %d; line should be in KEY=VALUE format", lineNumber)
```

File: cmd/main.go [hunk 2]
```diff
@@ -339,6 +340,73 @@ func flagHelp(fs *flag.FlagSet) string {
...
+		}
+
+		if strings.Contains(fields[0], " ") {
+			return nil, fmt.Errorf("Can't parse line %d; KEY contains whitespace", lineNumber)
```
Comment (mholt) [inline cmd/main.go:391]:
```suggestion
			return nil, fmt.Errorf("bad key on line %d: contains whitespace", lineNumber)
```

File: cmd/main.go [hunk 3]
```diff
@@ -339,6 +340,73 @@ func flagHelp(fs *flag.FlagSet) string {
...
+		}
+
+		key := fields[0]
+		val := fields[1]
+
+		if key == "" {
+			return nil, fmt.Errorf("Can't parse line %d; KEY can't be empty string", lineNumber)
```
Comment (mholt) [inline cmd/main.go:398]:
```suggestion
			return nil, fmt.Errorf("missing or empty key on line %d", lineNumber)
```

**Comment by elcore** [comment]
> elcore dismissed francislavoie’s stale review via a78e45a 9 minutes ago

No idea what I have done to dismiss the review... I have just resolved the merge conflict...

**Comment by elcore** [comment]
> we require any changes going into master to be reviewed

I know 😊

**Comment by jesse996** [comment]
could you add envfile in `caddy start` ?

**Comment by mholt** [comment]
@jesse996 Probably -- a bit busy right now, but PRs welcome, should be pretty easy!

**Pull request merged**